### PR TITLE
Subproject support

### PR DESCRIPTION
**Pull request by famz**
This is the minimum to handle subprojects. A subproject is basically a separate project but is better managed hierarchically to avoid clutter.

This should make it possible to fix the failed git applies in http://patchew.org/Libvirt/, where patches tagged as `python`, `tck`, `perl` etc. should acutally be applied on top of different git repos, instead of the main one.

### CODE REVIEW

File: mods/git.py [hunk 1]
```diff
@@ -141,19 +141,33 @@ def prepare_message_hook(self, request, message, detailed):
...
+                        message.extra_status.append({
+                            "kind": "good",
+                            "html": format_html('Patches applied successfully (<a href="{}">tree</a>, {}).<br/><samp>git fetch {} {}</samp>',
+                                                git_url, colorbox_a, git_repo, git_tag),
+                            "extra": colorbox_div,
+                            "id": "gitlog"
+                        })
+                else:
+                    message.status_tags.append({
+                        "title": format_html("Patch apply succeeded"),
```
Comment (bonzini) [inline mods/git.py]: "Patches applied successfully"?

**Comment by famz** [comment]
> I'm not sure I understand how there would be no repo/tag/URL, but the patches could still be applied successfully?

`patchew-cli apply --applier-mode` will still try to apply the series even when the project doesn't have `git.{push_to,public_repo,url_template}` properties, then save the result and log. Technically, as long as the `Project.git` field is correct, `git am` can be attempted.

This is the case on current next.patchew.org, and in general makes testing applier code easier. Although I agree this shouldn't exist in a real configuration.

**Comment by famz** [comment]
BTW I took down next.patchew.org last week due to a server error. Will re-deploy with current master as the Ansible patches are merged, after the Chinese New Year.

**Comment by bonzini** [comment]
Oh, that makes sense then!

File: mods/git.py [hunk 2]
```diff
@@ -141,19 +141,33 @@ def prepare_message_hook(self, request, message, detailed):
...
+                        "type": "info",
+                        "char": "G",
+                        })
+                    message.extra_status.append({
+                        "kind": "good",
+                        "html": format_html('Patches applied successfully'),
```
Comment (bonzini) [inline mods/git.py]: Please link the log here, like `format_html('Patches applied successfully ({})' , colorbox_a)`.

**Comment by famz** [comment]
Made the changes, but I propose we move to a mailing list + patchew.org workflow:

https://www.freelists.org/post/patchew-devel/PATCH-v3-02-Add-subproject-support

https://patchew.org/Patchew/20180222143731.21595-1-famz@redhat.com/

**Pull request closed without merging**